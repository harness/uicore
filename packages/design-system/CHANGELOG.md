# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# 1.4.0 (2022-07-04)


### Bug Fixes

* [CDS-39415]: added more colors in background prop ([#742](https://github.com/harness/uicore/issues/742)) ([a7ded54](https://github.com/harness/uicore/commit/a7ded54909efca0a3e71d223e5e1f57d1e2d84fd))





# [1.3.0](https://github.com/harness/uicore/compare/@harness/design-system@1.2.0...@harness/design-system@1.3.0) (2022-06-14)

**Note:** Version bump only for package @harness/design-system





# [1.2.0](https://github.com/harness/uicore/compare/@harness/design-system@1.1.0...@harness/design-system@1.2.0) (2022-06-08)


### Features

* [PL-25757]: Help Panel bugs fixes and changed exports in package json ([#702](https://github.com/harness/uicore/issues/702)) ([321e38d](https://github.com/harness/uicore/commit/321e38dc9b3399e1fa760ae07288ea18461e8cee))





# 1.1.0 (2022-06-06)


### Features

* [PL-22570]: Moving design-system to a separate package in UI-core ([#590](https://github.com/harness/uicore/issues/590)) ([8600cb6](https://github.com/harness/uicore/commit/8600cb6aa9d24531f1518b326abe2b2fe2440b98))
* [PL-24394]: Help Panel ([#646](https://github.com/harness/uicore/issues/646)) ([aa8c7e4](https://github.com/harness/uicore/commit/aa8c7e44d75b409ac9ea46b2538a78a2c8f291ad))
* Lerna Integration ([#685](https://github.com/harness/uicore/issues/685)) ([91caea1](https://github.com/harness/uicore/commit/91caea18921ec01266eb37b83d023612f9b41649))





# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.
