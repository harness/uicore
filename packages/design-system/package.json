{
  "name": "@harness/design-system",
  "version": "1.5.2",
  "description": "Harness Help panel - used for building Help panels generated through contentful",
  "source": "src/index.ts",
  "main": "./dist/design-system.umd.js",
  "exports": {
    ".": "./dist/design-system.es.js",
    "./dist/style.css": "./dist/style.css"
  },
  "types": "./dist/src/index.d.ts",
  "files": [
    "dist/*"
  ],
  "scripts": {
    "prebuild": "yarn clean",
    "build": "vite build",
    "clean": "rm -rf dist",
    "prepublishOnly": "yarn build",
    "test": "jest --no-cache src --config ../../jest.config.js"
  },
  "license": "UNLICENSED",
  "repository": "https://github.com/harness/uicore",
  "publishConfig": {
    "registry": "https://npm.pkg.github.com/"
  },
  "peerDependencies": {
    "react": "^17.0.2"
  },
  "devDependencies": {
    "@blueprintjs/core": "^3.20.0",
    "@types/react": "^17.0.3",
    "dts-bundle-generator": "^6.4.0",
    "react": "^17.0.2",
    "typescript": "^4.1.3",
    "vite": "^2.7.13",
    "vite-plugin-dts": "^0.9.9"
  }
}
