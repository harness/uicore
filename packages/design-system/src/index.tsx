/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

export * from './styled-props/StyledProps'
export * from './common/Color'
export * from './common/Spacing'
export * from './common/Types'
export * from './common/Intent'
export * from './styled-props/StyledProps'
export * from './styled-props/border/BorderProps'
export * from './styled-props/padding/PaddingProps'
export * from './styled-props/flex/FlexProps'
export * from './styled-props/font/FontProps'
export * from './styled-props/margin/MarginProps'

import styledClass from './styled-props/StyledProps.module.css'

export { styledClass }
