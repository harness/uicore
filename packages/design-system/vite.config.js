/*
 * Copyright 2020 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */
const path = require('path')
const _ = require('lodash')
const packages = require('./package.json')
const globals = require('../globals.json')
const dts = require('vite-plugin-dts')
import { createHash } from 'crypto'

const external = Object.keys(packages.peerDependencies)

external.forEach(dep => {
  if (!_.has(globals, dep)) {
    console.log(`Entry for "${dep}" not found in globals.json`)
    process.exit(1)
  }
})

/**
 * @type {import('vite').UserConfig}
 */
const config = {
  build: {
    lib: {
      entry: path.resolve(__dirname, 'src'),
      name: 'HarnessCore'
    },
    rollupOptions: {
      // make sure to externalize deps that shouldn't be bundled
      // into your library
      external,
      output: {
        // Provide global variables to use in the UMD build
        // for externalized deps
        globals: _.pick(globals, external)
      }
    }
  },
  css: {
    modules: {
      scopeBehaviour: 'local',
      generateScopedName: (name, filename, _css) => {
        const basename = path.basename(filename).replace(/\.module\.css?.*/, '')
        const hash = createHash('md5')
          .update(name + filename)
          .digest('hex')
          .slice(0, 5)
        return `${basename}--${name}--${hash}`
      }
    }
  },
  plugins: [dts()]
}

module.exports = config
