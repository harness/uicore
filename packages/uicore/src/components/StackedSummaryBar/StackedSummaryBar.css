/*
 * Copyright (c) Harness Inc.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

.summaryCount {
  width: var(--spacing-10);
  text-align: right;
  white-space: nowrap !important;
  flex-shrink: 0;
  flex-grow: 0;
}

.barSection {
  height: 8px;
  border-radius: 2px;
  margin-right: 1px !important;
}
