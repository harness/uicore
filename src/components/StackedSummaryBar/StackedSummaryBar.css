/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.summaryCount {
  width: 30px;
  text-align: right;
}

.barSection {
  height: 8px;
  border-radius: 2px;
  margin-right: 1px !important;
}
