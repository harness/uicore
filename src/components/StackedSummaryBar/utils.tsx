/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'

import { Color } from '@harness/design-system'
import { Container } from '../Container/Container'
import { Icon } from '../../icons/Icon'
import { Text } from '../Text/Text'

export const handleZeroOrInfinityTrend = (trend: string | undefined, intentColor: Color): JSX.Element => {
  const trendChange = trend ? parseInt(trend) : 0
  return isNaN(trendChange) ? ( // handling Infinity and other unexpected trends
    <Container flex={{ alignItems: 'center' }}>
      <Icon name={'caret-up'} color={intentColor}></Icon>
      <Text font="xsmall" color={intentColor}>
        N/A
      </Text>
    </Container>
  ) : (
    // handing 0 trend change
    <Container flex>{renderTrend(trend, Color.GREY_300)}</Container>
  )
}

export const renderTrend = (trend: string | undefined, intentColor: Color): JSX.Element => {
  const trendChange = trend ? parseInt(trend) : 0
  return (
    <>
      <Icon
        name={trendChange === 0 ? 'caret-right' : trendChange < 0 ? 'caret-down' : 'caret-up'}
        color={intentColor}></Icon>
      <Text font="xsmall" color={intentColor}>
        {Math.sign(trendChange) === -1 ? trend?.substring(1) : trend}
      </Text>
    </>
  )
}
