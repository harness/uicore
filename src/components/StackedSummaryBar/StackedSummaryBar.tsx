/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import css from './StackedSummaryBar.css'
import { Container } from '../Container/Container'
import { Color, Intent } from '@harness/design-system'
import { Layout } from '../../layouts/Layout'
import { Text } from '../Text/Text'
import { handleZeroOrInfinityTrend, renderTrend } from './utils'
import { Position } from '@blueprintjs/core'

export interface StackedBarSectionData {
  count: number
  color: string
}

export interface BarSection {
  width: number
  color: string
}

export interface StackedSummaryBarData {
  barSectionsData: StackedBarSectionData[]
  trend?: string
  intent?: Intent
  showTrend?: boolean
}

export interface StackedSummaryBarProps extends StackedSummaryBarData {
  maxCount: number
  barLength?: number
}

export const getStackedSummaryBarCount = (barData: StackedBarSectionData[]): number =>
  barData.reduce((sum: number, section: StackedBarSectionData) => sum + section.count, 0)

export const StackedSummaryBar: React.FC<StackedSummaryBarProps> = props => {
  const { maxCount, barSectionsData, trend, intent = Intent.SUCCESS, barLength = 100, showTrend = true } = props
  const summaryCount = getStackedSummaryBarCount(barSectionsData)
  const trendChange = trend ? parseInt(trend) : 0
  const barSections: BarSection[] = []
  const effectiveBarLength = barLength - barSectionsData.length // for 1px gaps

  // Can not have stackBar without these data
  if (!(maxCount > 0) || !barSectionsData.length) {
    return null
  }

  barSectionsData.forEach((stackedBarSection: StackedBarSectionData) => {
    stackedBarSection.count && // section for 0 count is not required
      barSections.push({
        width: (stackedBarSection.count / maxCount) * effectiveBarLength,
        color: stackedBarSection.color
      })
  })
  // leftover section relative to maxCount should be as blank
  barSections.push({ width: (1 - summaryCount / maxCount) * effectiveBarLength, color: Color.GREY_100 })

  const formattedSummaryCount = React.useMemo(() => {
    return new Intl.NumberFormat('default', {
      notation: 'compact',
      compactDisplay: 'short',
      unitDisplay: 'long',
      minimumFractionDigits: 0,
      maximumFractionDigits: 2
    }).format(summaryCount)
  }, [summaryCount])

  return (
    <Layout.Horizontal spacing="small">
      <Text
        font="small"
        className={css.summaryCount}
        tooltip={<Text padding={'small'}>{summaryCount}</Text>}
        alwaysShowTooltip={true}
        tooltipProps={{
          position: Position.TOP_RIGHT
        }}>
        {formattedSummaryCount}
      </Text>
      <Container flex>
        {barSections.map((barSection: BarSection, index: number) => {
          return barSection.width ? (
            <Container
              background={barSection.color}
              key={index}
              className={css.barSection}
              width={barSection.width}></Container>
          ) : null
        })}
      </Container>
      {showTrend ? (
        trendChange ? (
          <Container flex>
            {(intent === Intent.SUCCESS) === trendChange < 0
              ? renderTrend(trend, Color.RED_500)
              : renderTrend(trend, Color.GREEN_500)}
          </Container>
        ) : (
          handleZeroOrInfinityTrend(trend, intent === Intent.SUCCESS ? Color.GREEN_500 : Color.RED_500)
        )
      ) : (
        <></>
      )}
    </Layout.Horizontal>
  )
}
