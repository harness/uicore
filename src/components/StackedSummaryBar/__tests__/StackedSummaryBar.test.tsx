/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import { Color } from '@harness/design-system'
import { StackedSummaryBar } from '../StackedSummaryBar'

const barData = [
  { color: Color.GREEN_500, count: 50 },
  { color: Color.RED_500, count: 10 }
]

describe('StackedSummaryBar', () => {
  test('Stack without trend', async () => {
    const { container } = render(<StackedSummaryBar barSectionsData={barData} maxCount={100} />)
    expect(container).toMatchSnapshot()
  })

  test('Stack with positive trend', async () => {
    const { container } = render(<StackedSummaryBar barSectionsData={barData} maxCount={100} trend={'10%'} />)
    expect(container).toMatchSnapshot()
  })

  test('Stack with negative trend', async () => {
    const { container } = render(<StackedSummaryBar barSectionsData={barData} maxCount={100} trend={'-10%'} />)
    expect(container).toMatchSnapshot()
  })

  test('Stack with 0 trend', async () => {
    const { container } = render(<StackedSummaryBar barSectionsData={barData} maxCount={100} trend={'0%'} />)
    expect(container).toMatchSnapshot()
  })

  test('Stack with infinity trend', async () => {
    const { container } = render(<StackedSummaryBar barSectionsData={barData} maxCount={100} trend={'infinity%'} />)
    expect(container).toMatchSnapshot()
  })
})
