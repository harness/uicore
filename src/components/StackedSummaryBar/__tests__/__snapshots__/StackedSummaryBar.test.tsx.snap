// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StackedSummaryBar Stack with 0 trend 1`] = `
<div>
  <div
    class="horizontal layout-spacing-small StyledProps--main"
  >
    <span
      class="bp3-popover-wrapper"
    >
      <span
        class="bp3-popover-target targetClass"
      >
        <p
          class="StyledProps--font StyledProps--main summaryCount StyledProps--font-small"
          tabindex="0"
        >
          60
        </p>
      </span>
    </span>
    <div
      class="StyledProps--font StyledProps--main StyledProps--flex"
    >
      <div
        class="StyledProps--font StyledProps--main barSection StyledProps--background-green500"
        style="width: 49px;"
      />
      <div
        class="StyledProps--font StyledProps--main barSection StyledProps--background-red500"
        style="width: 9.8px;"
      />
      <div
        class="StyledProps--font StyledProps--main barSection StyledProps--background-grey100"
        style="width: 39.2px;"
      />
    </div>
    <div
      class="StyledProps--font StyledProps--main StyledProps--flex"
    >
      <span
        class="bp3-icon bp3-icon-caret-right StyledProps--main StyledProps--color StyledProps--color-grey300"
        icon="caret-right"
      >
        <svg
          data-icon="caret-right"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <desc>
            caret-right
          </desc>
          <path
            d="M11 8c0-.15-.07-.28-.17-.37l-4-3.5A.495.495 0 006 4.5v7a.495.495 0 00.83.37l4-3.5c.1-.09.17-.22.17-.37z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-xsmall StyledProps--color StyledProps--color-grey300"
      >
        0%
      </p>
    </div>
  </div>
</div>
`;

exports[`StackedSummaryBar Stack with infinity trend 1`] = `
<div>
  <div
    class="horizontal layout-spacing-small StyledProps--main"
  >
    <span
      class="bp3-popover-wrapper"
    >
      <span
        class="bp3-popover-target targetClass"
      >
        <p
          class="StyledProps--font StyledProps--main summaryCount StyledProps--font-small"
          tabindex="0"
        >
          60
        </p>
      </span>
    </span>
    <div
      class="StyledProps--font StyledProps--main StyledProps--flex"
    >
      <div
        class="StyledProps--font StyledProps--main barSection StyledProps--background-green500"
        style="width: 49px;"
      />
      <div
        class="StyledProps--font StyledProps--main barSection StyledProps--background-red500"
        style="width: 9.8px;"
      />
      <div
        class="StyledProps--font StyledProps--main barSection StyledProps--background-grey100"
        style="width: 39.2px;"
      />
    </div>
    <div
      class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-alignItems-center"
    >
      <span
        class="bp3-icon bp3-icon-caret-up StyledProps--main StyledProps--color StyledProps--color-green500"
        icon="caret-up"
      >
        <svg
          data-icon="caret-up"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <desc>
            caret-up
          </desc>
          <path
            d="M11.87 9.17s.01 0 0 0l-3.5-4C8.28 5.07 8.15 5 8 5s-.28.07-.37.17l-3.5 4a.495.495 0 00.37.83h7a.495.495 0 00.37-.83z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-xsmall StyledProps--color StyledProps--color-green500"
      >
        N/A
      </p>
    </div>
  </div>
</div>
`;

exports[`StackedSummaryBar Stack with negative trend 1`] = `
<div>
  <div
    class="horizontal layout-spacing-small StyledProps--main"
  >
    <span
      class="bp3-popover-wrapper"
    >
      <span
        class="bp3-popover-target targetClass"
      >
        <p
          class="StyledProps--font StyledProps--main summaryCount StyledProps--font-small"
          tabindex="0"
        >
          60
        </p>
      </span>
    </span>
    <div
      class="StyledProps--font StyledProps--main StyledProps--flex"
    >
      <div
        class="StyledProps--font StyledProps--main barSection StyledProps--background-green500"
        style="width: 49px;"
      />
      <div
        class="StyledProps--font StyledProps--main barSection StyledProps--background-red500"
        style="width: 9.8px;"
      />
      <div
        class="StyledProps--font StyledProps--main barSection StyledProps--background-grey100"
        style="width: 39.2px;"
      />
    </div>
    <div
      class="StyledProps--font StyledProps--main StyledProps--flex"
    >
      <span
        class="bp3-icon bp3-icon-caret-down StyledProps--main StyledProps--color StyledProps--color-red500"
        icon="caret-down"
      >
        <svg
          data-icon="caret-down"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <desc>
            caret-down
          </desc>
          <path
            d="M12 6.5c0-.28-.22-.5-.5-.5h-7a.495.495 0 00-.37.83l3.5 4c.09.1.22.17.37.17s.28-.07.37-.17l3.5-4c.08-.09.13-.2.13-.33z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-xsmall StyledProps--color StyledProps--color-red500"
      >
        10%
      </p>
    </div>
  </div>
</div>
`;

exports[`StackedSummaryBar Stack with positive trend 1`] = `
<div>
  <div
    class="horizontal layout-spacing-small StyledProps--main"
  >
    <span
      class="bp3-popover-wrapper"
    >
      <span
        class="bp3-popover-target targetClass"
      >
        <p
          class="StyledProps--font StyledProps--main summaryCount StyledProps--font-small"
          tabindex="0"
        >
          60
        </p>
      </span>
    </span>
    <div
      class="StyledProps--font StyledProps--main StyledProps--flex"
    >
      <div
        class="StyledProps--font StyledProps--main barSection StyledProps--background-green500"
        style="width: 49px;"
      />
      <div
        class="StyledProps--font StyledProps--main barSection StyledProps--background-red500"
        style="width: 9.8px;"
      />
      <div
        class="StyledProps--font StyledProps--main barSection StyledProps--background-grey100"
        style="width: 39.2px;"
      />
    </div>
    <div
      class="StyledProps--font StyledProps--main StyledProps--flex"
    >
      <span
        class="bp3-icon bp3-icon-caret-up StyledProps--main StyledProps--color StyledProps--color-green500"
        icon="caret-up"
      >
        <svg
          data-icon="caret-up"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <desc>
            caret-up
          </desc>
          <path
            d="M11.87 9.17s.01 0 0 0l-3.5-4C8.28 5.07 8.15 5 8 5s-.28.07-.37.17l-3.5 4a.495.495 0 00.37.83h7a.495.495 0 00.37-.83z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-xsmall StyledProps--color StyledProps--color-green500"
      >
        10%
      </p>
    </div>
  </div>
</div>
`;

exports[`StackedSummaryBar Stack without trend 1`] = `
<div>
  <div
    class="horizontal layout-spacing-small StyledProps--main"
  >
    <span
      class="bp3-popover-wrapper"
    >
      <span
        class="bp3-popover-target targetClass"
      >
        <p
          class="StyledProps--font StyledProps--main summaryCount StyledProps--font-small"
          tabindex="0"
        >
          60
        </p>
      </span>
    </span>
    <div
      class="StyledProps--font StyledProps--main StyledProps--flex"
    >
      <div
        class="StyledProps--font StyledProps--main barSection StyledProps--background-green500"
        style="width: 49px;"
      />
      <div
        class="StyledProps--font StyledProps--main barSection StyledProps--background-red500"
        style="width: 9.8px;"
      />
      <div
        class="StyledProps--font StyledProps--main barSection StyledProps--background-grey100"
        style="width: 39.2px;"
      />
    </div>
    <div
      class="StyledProps--font StyledProps--main StyledProps--flex"
    >
      <span
        class="bp3-icon bp3-icon-caret-right StyledProps--main StyledProps--color StyledProps--color-grey300"
        icon="caret-right"
      >
        <svg
          data-icon="caret-right"
          height="16"
          viewBox="0 0 16 16"
          width="16"
        >
          <desc>
            caret-right
          </desc>
          <path
            d="M11 8c0-.15-.07-.28-.17-.37l-4-3.5A.495.495 0 006 4.5v7a.495.495 0 00.83.37l4-3.5c.1-.09.17-.22.17-.37z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      <p
        class="StyledProps--font StyledProps--main StyledProps--font-xsmall StyledProps--color StyledProps--color-grey300"
      />
    </div>
  </div>
</div>
`;
